### PR TITLE
priority partitioner

### PR DESCRIPTION
**Pull request by guysegal**

### CODE REVIEW

File: service.log
Comment (omerlh) [inline service.log:1]: remove + add to .gitignore

File: src/main/scala/partitioners/GroupPartitioner.scala [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package partitioners
+
+import org.apache.kafka.clients.producer.Partitioner
+import org.apache.kafka.common.Cluster
+
+class GroupPartitioner extends Partitioner {
+  private val partitionsPerGroup = 5
```
Comment (omerlh) [inline src/main/scala/partitioners/GroupPartitioner.scala]: You need to read it from the config

**Review by omerlh — APPROVED** [review]

File: src/main/scala/partitioners/GroupPartitioner.scala (hunk 1 above)
Comment (guysegal) [reply src/main/scala/partitioners/GroupPartitioner.scala]: I didn't understand how to do that so I put const in the meanwhile

File: src/main/scala/config/KafkaConfig.scala [hunk 2]
```diff
@@ -76,9 +77,12 @@ object KafkaConfig {
           ("value.serializer", "org.apache.kafka.common.serialization.StringSerializer"),
           ("bootstrap.servers", broker),
           ("linger.ms", lingerTime),
-          ("max.block.ms", maxBlockMS)
+          ("max.block.ms", maxBlockMS),
+          //todo: add condition here
+          ("partitioner.class", if (useGroupPartitioner) "partitioners.GroupPartitioner" else "org.apache.kafka.clients.producer.internals.DefaultPartitioner")
```
Comment (guysegal) [inline src/main/scala/config/KafkaConfig.scala]: @omerlh I would like to not add `partitioner.class` if useGroupPartitioner=false. also didn't figure out how

Comment (guysegal) [reply src/main/scala/config/KafkaConfig.scala]: I figured this one out

File: src/main/scala/partitioners/PriorityPartitioner.scala [hunk 3]
```diff
@@ -0,0 +1,32 @@
+package partitioners
+
+import org.apache.kafka.clients.producer.Partitioner
+import org.apache.kafka.common.Cluster
+import org.apache.kafka.common.utils.Utils
+
+class PriorityPartitioner extends Partitioner {
+  private val partitionsPerPriority = 2
```
Comment (guysegal) [inline src/main/scala/partitioners/PriorityPartitioner.scala:8]: @omerlh can you help me with syntax here? I don't get how to read from configuration, and I tried..

**Pull request closed without merging**